build: std::views::values not supported on macOS Sonoma 14.5 Apple Clang 14

Upgrading to macOS Sonoma version 14.5 while keeping Apple clang version 14.0.3 breaks our build:

```make
[ 39%] Built target sentry
[ 39%] Built target sentry_client_test
[ 55%] Built target torrent-rasterbar
[ 55%] Building CXX object silkworm/db/snapshots/CMakeFiles/silkworm_snapshots.dir/repository.cpp.o
In file included from /Users/tullio/Workspace/silkworm/silkworm/db/snapshots/repository.cpp:17:
In file included from /Users/tullio/Workspace/silkworm/silkworm/db/snapshots/repository.hpp:29:
/Users/tullio/Workspace/silkworm/silkworm/db/snapshots/common/iterator/map_values_view.hpp:84:12: error: no member named 'values' in namespace 'std::ranges::views'; did you mean 'std::money_base::value'?
    return std::views::values(map);
           ^~~~~~~~~~~~~~~~~~
           std::money_base::value
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/locale:2589:43: note: 'std::money_base::value' declared here
    enum part {none, space, symbol, sign, value};
                                          ^
In file included from /Users/tullio/Workspace/silkworm/silkworm/db/snapshots/repository.cpp:17:
In file included from /Users/tullio/Workspace/silkworm/silkworm/db/snapshots/repository.hpp:29:
/Users/tullio/Workspace/silkworm/silkworm/db/snapshots/common/iterator/map_values_view.hpp:84:12: error: called object type 'std::money_base::part' is not a function or function pointer
    return std::views::values(map);
           ^~~~~~~~~~~~~~~~~~
/Users/tullio/Workspace/silkworm/silkworm/db/snapshots/repository.hpp:74:40: note: in instantiation of function template specialization 'silkworm::make_map_values_view<unsigned long long, silkworm::snapshots::SnapshotBundle>' requested here
    auto view_bundles() const { return make_map_values_view(bundles_); }
                                       ^
/Users/tullio/Workspace/silkworm/silkworm/db/snapshots/repository.cpp:178:31: error: cannot use type 'void' as a range
    for (const auto& bundle : this->view_bundles_reverse()) {
                              ^
3 errors generated.
make[2]: *** [silkworm/db/snapshots/CMakeFiles/silkworm_snapshots.dir/repository.cpp.o] Error 1
make[1]: *** [silkworm/db/snapshots/CMakeFiles/silkworm_snapshots.dir/all] Error 2
make: *** [all] Error 2
```

--- a/silkworm/db/snapshots/common/iterator/map_values_view.hpp
+++ b/silkworm/db/snapshots/common/iterator/map_values_view.hpp
@@ -78,7 +78,9 @@ class MapValuesView : std::ranges::view_interface<MapValuesView<TMapKey, TMapVal
 template <typename TMapKey, typename TMapValue>
 auto make_map_values_view(const std::map<TMapKey, TMapValue>& map) {
     // std::views::values is not present on clang 15
-#if defined(__clang__) && (__clang_major__ <= 15)
+#if defined(__clang__) && (__clang_major__ <= 15) && !defined(__apple_build_version__)
+    return MapValuesView<TMapKey, TMapValue>{map};
+#elif defined(__clang__) && (__clang_major__ <= 14) && defined(__apple_build_version__)  // clang 15 == Apple clang 14
     return MapValuesView<TMapKey, TMapValue>{map};
 #else
     return std::views::values(map);

--- a/silkworm/db/snapshots/common/iterator/map_values_view.hpp
+++ b/silkworm/db/snapshots/common/iterator/map_values_view.hpp
@@ -78,7 +78,7 @@ class MapValuesView : std::ranges::view_interface<MapValuesView<TMapKey, TMapVal
 template <typename TMapKey, typename TMapValue>
 auto make_map_values_view(const std::map<TMapKey, TMapValue>& map) {
     // std::views::values is not present on clang 15
-#if defined(__clang__) && (__clang_major__ <= 15) && !defined(__apple_build_version__)
+#if defined(__clang__) && (__clang_major__ <= 15)
     return MapValuesView<TMapKey, TMapValue>{map};
 #else
     return std::views::values(map);

